src/frontend/src/planner/select.rs

```diff
@@ -413,6 +430,140 @@ impl Planner {
         Ok((root, exprs))
     }
 
+    /// Substitutes all [`Subquery`] in `exprs` in a cross join way.
+    ///
+    /// Each time a [`Subquery`] is found, it is replaced by a new [`InputRef`]. And `root` is
+    /// replaced by a new `LeftOuter` [`LogicalApply`] whose left side is `root` and right side is
+    /// the planned subquery.
+    ///
+    /// The [`InputRef`]s' indexes start from `root.schema().len()`,
+    /// which means they are additional columns beyond the original `root`.
+    ///
+    ///
+    /// The cross join way only meaningful when there are multiple subqueries
+    ///
+    /// ```text
+    ///            Apply
+    ///           /    \
+    ///         left   CrossJoin
+    ///                   /   \
+    ///           Subquery1   CrossJoin
+    ///                         /   \
+    ///                 Subquery2   Subquery3
+    /// ```
+    /// Typically, this is used for scalar subqueries in select clauses, because users might write subqueries in a case-when exprs multiple times.
```

Comment:
nit about the comment: The case doesn't need to involve case-when, e.g., `select subquery1, subquery2`. 

(I thought this is tricky, but in PG we cannot refer to the alias in the same SELECT, so seems not as hard as I imagined..)